lib/Backend/BailOut.cpp

```diff
@@ -2200,6 +2200,12 @@ void BailOutRecord::ScheduleFunctionCodeGen(Js::ScriptFunction * function, Js::S
                     rejitReason = RejitReason::PowIntIntTypeSpecDisabled;
                 }
             }
+            case IR::BailOutOnEarlyExit:
+            {
+                profileInfo->DisableOptimizeTryFinally();
```

Comment:
You should follow the same pattern as above:
if (already disabled)
   reThunk to the latest JIT'd code
else
   disable it